roxctl/central/crs/revoke.go

```diff
@@ -0,0 +1,92 @@
+package crs
+
+import (
+	"context"
+	"strings"
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+	v1 "github.com/stackrox/rox/generated/api/v1"
+	"github.com/stackrox/rox/pkg/errox"
+	pkgCommon "github.com/stackrox/rox/pkg/roxctl/common"
+	"github.com/stackrox/rox/pkg/set"
+	"github.com/stackrox/rox/pkg/utils"
+	"github.com/stackrox/rox/roxctl/common"
+	"github.com/stackrox/rox/roxctl/common/environment"
+	"github.com/stackrox/rox/roxctl/common/flags"
+	"github.com/stackrox/rox/roxctl/common/logger"
+)
+
+func applyRevokeCRSs(ctx context.Context, cliEnvironment environment.Environment, svc v1.ClusterInitServiceClient, idsOrNames set.StringSet) error {
+	resp, err := svc.GetCRSs(ctx, &v1.Empty{})
+	if err != nil {
+		return err
+	}
+
+	var revokeIds []string
+	for _, meta := range resp.Items {
+		if idsOrNames.Remove(meta.GetId()) || idsOrNames.Remove(meta.GetName()) {
+			revokeIds = append(revokeIds, meta.GetId())
+		}
+	}
+
+	if len(idsOrNames) != 0 {
+		return errox.NotFound.Newf("could not find CRS(s) %s", strings.Join(idsOrNames.AsSlice(), ", "))
+	}
+
+	revokeResp, err := svc.RevokeCRS(ctx, &v1.CRSRevokeRequest{Ids: revokeIds})
+	if err != nil {
+		return errors.Wrap(err, "revoking CRSs")
+	}
+	printResponseResult(cliEnvironment.Logger(), revokeIds, revokeResp)
+
+	return nil
+}
+
+func printResponseResult(logger logger.Logger, revokeIds []string, resp *v1.CRSRevokeResponse) {
+	for _, id := range resp.GetRevokedIds() {
+		logger.InfofLn("Revoked %q", id)
```

Comment:
nitpick: I'm wondering if we could improve the information here and avoid confusion. Because a user can provide a CRS name, and the user will see only the ID of the removed CRS. It's not clear what happened there.

If we change `var revokeIds []string` - to `var revokeIds map[string]string` and store `id->name`, we could improve the log here and provide both information.

Reply:
done